src/back/spv/writer.rs

```diff
@@ -1330,7 +1338,53 @@ impl Writer {
                             ));
                             id
                         }
-                        //TODO: support `crate::TypeInner::Array { .. }` ?
+                        crate::TypeInner::Array { base: ty_base, .. } => {
+                            // promote the expression into a variable
+                            let base_type_id =
+                                self.get_expression_type_id(&ir_module.types, &fun_info[base].ty)?;
+                            let pointer_type_id = self.id_gen.next();
+                            Instruction::type_pointer(
+                                pointer_type_id,
+                                spirv::StorageClass::Function,
+                                base_type_id,
+                            )
+                            .to_words(&mut self.logical_layout.declarations);
+                            let variable_id = self.id_gen.next();
+                            let instruction = Instruction::variable(
+                                pointer_type_id,
+                                variable_id,
+                                spirv::StorageClass::Function,
+                                None,
+                            );
+                            function.internal_variables.push(LocalVariable {
+                                id: variable_id,
+                                instruction,
+                            });
+
+                            let element_pointer_id = self.id_gen.next();
+                            let element_pointer_type_id = self.get_pointer_id(
+                                &ir_module.types,
+                                ty_base,
+                                spirv::StorageClass::Function,
+                            )?;
+                            block
+                                .body
+                                .push(Instruction::store(variable_id, base_id, None));
```

Comment:
Nit: It would be a bit clearer to push this instruction immediately after registering the `LocalVariable`, so that `element_pointer_id` is generated directly above its use.